### PR TITLE
Fixes #15202: Logging the remote-run API with warp

### PR DESCRIPTION
**Pull request by amousset**

### CODE REVIEW

File: relay/sources/relayd/src/api.rs [hunk 1]
```diff
@@ -52,21 +52,19 @@ pub fn api(
...
-        warp::reply::json(&Status::poll(job_config1.clone()))
-    });
+    let status =
+        warp::path("status").map(move || warp::reply::json(&Status::poll(job_config1.clone())));
 
     let relay_api = warp::path("rudder").and(warp::path("relay-api"));
     let remote_run = relay_api.and(warp::path("remote-run"));
     let shared_files = relay_api.and(warp::path("shared-files"));
 
+    let log = warp::log("relayd::relay-api");
```
Comment (amousset) [inline relay/sources/relayd/src/api.rs]: no need for the log variable, it can go directly into the `with` call

**Comment by ArmandSy** [comment]
Commit modified

File: relay/sources/relayd/src/api.rs [hunk 2]
```diff
@@ -107,45 +103,30 @@ pub fn api(
...
     let routes = warp::get2()
         .and(status.or(stats_simple))
         .or(warp::post2().and(remote_run).and(nodes.or(all).or(node_id)))
         .or(warp::put2().and(shared_files).and(shared_files_put))
-        .or(warp::head().and(shared_files).and(shared_files_head));
+        .or(warp::head().and(shared_files).and(shared_files_head))
+        .with(warp::log("relayd::relay-api"));
 
     let (addr, server) = warp::serve(routes).bind_with_graceful_shutdown(listen, shutdown);
-    info!("Started stats API on {}", addr);
```
Comment (amousset) [inline relay/sources/relayd/src/api.rs]: This one is useful as it gives the information the API is started, and the URL to access it

Comment (amousset) [reply relay/sources/relayd/src/api.rs]: But it's not a stats API anymore, so the message could simply be `Started API on {}`

**Comment by ArmandSy** [comment]
Commit modified

**Pull request merged**